todos-app/src/App/TodoList/TodoList.tsx

```diff
@@ -1,40 +1,46 @@
-import React, { ReactElement, useState } from "react";
-import { ITodo } from "../../DataStructure";
+import { useEffect, useState } from "react";
+import { Todo } from "../models/Todos";
 import TodoListItem from "./TodoListItem/TodoListItem";
 
-type TodoListProps = {
-  todos: ITodo[];
+type Props = {
+  todos: Todo[];
   onToggle(id: string): void;
   onRemove(id: string): void;
 };
 
-const TodoList: React.FC<TodoListProps> = ({ todos, onToggle, onRemove }) => {
-  const toggleAllCheckbox = (): void => {
+const TodoList = ({ todos, onToggle, onRemove }: Props) => {
+  const [isChecked, setIsChecked] = useState(false);
+  let toggleAllState = todos.every((todo) => {
+    return todo.completed === true;
+  });
+
+  useEffect(() => {
     todos.forEach((todo) => {
-      onToggle(todo.id);
+      if (todo.completed === toggleAllState) {
+        onToggle(todo.id);
+      }
     });
-  };
+  }, [isChecked]);
 
   return (
     <section className="main">
       <input
         id="toggle-all"
         className="toggle-all"
         type="checkbox"
-        onChange={toggleAllCheckbox}
+        checked={toggleAllState}
+        onChange={() => setIsChecked(!isChecked)}
```

Comment:
it's a bit hacky to use state and effect to trigger some action, so rather than setting state and then using effect to trigger toggle, you can define a function for the onChange handler that can do directly

example:
```
 const handleCheckboxChange = (event) => {
  const isChecked = event.target.checked;
  console.log(isChecked);
};

<input type="checkbox" onChange={handleCheckboxChange} />

```

Reply:
Ok, now I'm get it. I'll fix it today